src/main/java/hexgui/gui/HexGui.java

```diff
@@ -0,0 +1,2393 @@
+    JFrame frame = new JFrame();
+    // frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+    Container con = frame.getContentPane();
+
+    PrintPreview pp = new PrintPreview(m_guiboard);
+    con.add(pp, BorderLayout.CENTER);
+
+    frame.pack();
+    frame.setVisible(true);
+    frame.toFront();
+  }
+
+  private void cmdPrint() {
+    Print.run(this, m_guiboard);
+  }
+
+  private void cmdAbout() {
+    m_about.setVisible(true);
+  }
+
+  // ------------------------------------------------------------
+
+  private void cmdGuiToolbarVisible() {
+    boolean visible = m_menubar.getToolbarVisible();
+    m_toolbar.setVisible(visible);
+  }
+
+  private void cmdGuiShellVisible() {
+    if (m_shell == null) return;
+    boolean visible = m_menubar.getShellVisible();
+    m_shell.setVisible(visible);
+  }
+
+  private void cmdGuiAnalyzeVisible() {
+    if (m_analyzeDialog == null) return;
+    boolean visible = m_menubar.getAnalyzeVisible();
+    m_analyzeDialog.setVisible(visible);
+  }
+
+  private void cmdGuiBoardDrawType() {
+    String type = m_menubar.getCurrentBoardDrawType();
+    System.out.println(type);
+    m_guiboard.setDrawType(type);
+    m_guiboard.repaint();
+  }
+
+  private void cmdGuiBoardOrientation() {
+    String type = m_menubar.getCurrentBoardOrientation();
+    System.out.println(type);
+    m_guiboard.setOrientation(type);
+    m_guiboard.repaint();
+  }
+
+  private void cmdClearMarks() {
+    m_guiboard.clearMarks();
+    m_guiboard.repaint();
+  }
+
+  private void cmdShowPreferences() {
+    new PreferencesDialog(this, m_preferences);
+  }
+
+  /**
+   * Toggle the player to move, by explicit user request. This also updates the PL property in the
+   * current node.
+   */
+  private void cmdToggleToMove() {
+    this.toggleToMove();
+    m_current.setPlayerToMove(m_tomove);
+  }
+
+  /** Toggle the player to move, without setting the PL property */
+  private void toggleToMove() {
+    m_tomove = m_tomove.otherColor();
+    m_toolbar.setToMove(m_tomove.toString());
+    m_menubar.setToMove(m_tomove.toString());
+    setCursorType();
+  }
+
+  /**
+   * Set the player to move, by explicit user request. This also updates the PL property in the
+   * current node.
+   */
+  private void cmdSetToMove() {
+    this.setToMove();
+    m_current.setPlayerToMove(m_tomove);
+  }
+
+  /** Set the player to move, without setting the PL property */
+  private void setToMove() {
+    m_tomove = HexColor.get(m_menubar.getToMove());
+    m_toolbar.setToMove(m_tomove.toString());
+    setCursorType();
+  }
+
+  private void cmdSetupBlack() {
+    setCursorType();
+  }
+
+  private void cmdSetupWhite() {
+    setCursorType();
+  }
+
+  // Update the cursor according to the current move type.
+  public void setCursorType() {
+    String clickContext = m_toolbar.getClickContext();
+    if (clickContext == "black") {
+      m_guiboard.setCursorType("black-setup");
+    } else if (clickContext == "white") {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.